### PR TITLE
feat(ripple): Implement subset of improved interaction response guide…

### LINKED ISSUE #190: Implement subset of improved interaction response guidelines for ripple
-  _Tap_ - This could be a quick mousedown/mouseup, quick keydown/keyup, quick pointerstart/pointerend, trackpad tap, etc.
- _Tap and hold_ - Mousedown/keydown/pointerstart + hold + mouseup/keyup/pointerend

The rest of the improved interaction response guidelines will be covered in separate stories.

### PR DESCRIPTION
**Pull request by cristobalchao**
feat(ripple): Implement subset of improved interaction response guidelines for ripple

Resolves #190

### CODE REVIEW

**Comment by cristobalchao** [comment]
Ups, it seemed I got rid of that failure but it showed up again...

**Comment by cristobalchao** [comment]
All issues addressed PTAL

File: packages/mdc-ripple/foundation.js [hunk 1]
```diff
@@ -146,12 +148,20 @@ export default class MDCRippleFoundation extends MDCFoundation {
 
     activationState.isActivated = true;
     activationState.isProgrammatic = e === null;
+    activationState.isTimeoutDeactivated = false;
```
Comment (traviskaufman) [inline packages/mdc-ripple/foundation.js]: I think this, as well as `isProgrammatic`, need to be added to the return value of `defaultActivationState_()`

File: packages/mdc-ripple/foundation.js [hunk 2]
```diff
@@ -146,12 +148,20 @@ export default class MDCRippleFoundation extends MDCFoundation {
...
     activationState.activationEvent = e;
     activationState.wasActivatedByPointer = activationState.isProgrammatic ? false : (
       e.type === 'mousedown' || e.type === 'touchstart' || e.type === 'pointerdown'
     );
 
     activationState.activationStartTime = Date.now();
+
+    activationState.deactivationTimeout = setTimeout(() => {
```
Comment (traviskaufman) [inline packages/mdc-ripple/foundation.js]: same comment as above

**Comment by cristobalchao** [comment]
All issues addressed PTAL

File: test/unit/mdc-ripple/foundation-deactivation.test.js [hunk 3]
```diff
@@ -101,6 +101,43 @@ testFoundation('runs deactivation UX on public deactivate() call', ({foundation,
...
+      ({foundation, adapter, mockRaf}) => {
+  const handlers = captureHandlers(adapter);
+  foundation.init();
+  mockRaf.flush();
+
+  handlers.mousedown();
+  mockRaf.flush();
+
+  setTimeout(() => {
+    td.verify(adapter.removeClass(cssClasses.BG_ACTIVE));
```
Comment (traviskaufman) [inline test/unit/mdc-ripple/foundation-deactivation.test.js]: Not exactly sure how these tests are different from the ones above...

Should this be verifying that these adapter methods are _not_ called _before_ `DEACTIVATION_TIMEOUT_MS`?

**Comment by cristobalchao** [comment]
All issues addressed PTAL
View changes

File: packages/mdc-ripple/constants.js [hunk 4]
```diff
@@ -53,4 +53,5 @@ export const numbers = {
   UNBOUNDED_TRANSFORM_DURATION_MS: 200,
   PADDING: 10,
   INITIAL_ORIGIN_SCALE: 0.6,
+  DEACTIVATION_TIMEOUT_MSL: 200,
```
Comment (traviskaufman) [inline packages/mdc-ripple/constants.js]: looks like this should be named `DEACTIVATION_TIMEOUT_MSL` instead of `DEACTIVATION_TIMEOUT_MS`

**Pull request closed without merging**